src/lib.rs

```diff
@@ -5,6 +5,7 @@ mod config;
 mod cri_service;
 mod criapi;
 mod image_service;
+mod network;
```

Comment:
Should we call this CNI?

Reply:
The question is at which point we want to abstract. My thought was that we use the term `network` for any network, which can be also non CNI-based ones (I was thinking about rootless implementations like slirp4netns). If so, then it would stack:

- network
  - cni
  - other

If we say we only want to support CNI, then I can change it to remove the whole network abstraction and flatten some structures. WDYT?